tool/m4/mmtk_ruby.m4

```diff
@@ -2,6 +2,7 @@ dnl -*- Autoconf -*-
 AC_DEFUN([MMTK_RUBY], [
 
 [MMTK_RUBY_SO_NAME=libmmtk_ruby.so]
+[MMTK_RUBY_DYLIB_NAME=libmmtk_ruby.dylib]
```

Comment:
I tried using this locally on macos and these changes threw an error since it was still trying to load the `so` file.

The Ruby build system already has code to replace the extension, this can use `$SOEXT` instead and then you don't need two separate variables.

From my testing, if you change `[MMTK_RUBY_SO_NAME=libmmtk_ruby.so]` to `[MMTK_RUBY_SO_NAME=libmmtk_ruby.$SOEXT]` you can build with MMTK on macos and it will use the `libmmtk_ruby.dylib`.